Sources/Sampler/py_sampler.hpp

```diff
@@ -152,6 +161,94 @@ void AddSamplerModule(py::module& m) {
   AddExactSampler(subm);
   AddCustomSampler(subm);
   AddMetropolisHastings(subm);
+
+
+  py::class_<MCResult>(subm, "MCResult",
+                       R"EOF(Result of Monte Carlo sampling.)EOF")
+      .def_property_readonly(
+          "samples",
+          [](const MCResult &self) {
+            assert(self.samples.rows() % self.n_chains == 0);
+            return detail::as_readonly(py::array_t<double, py::array::c_style>{
+                {self.samples.rows() / self.n_chains, self.n_chains,
+                 self.samples.cols()},
+                self.samples.data(),
+                py::none()});
+          },
+          py::return_value_policy::reference_internal,
+          R"EOF(Visible configurations `{vᵢ}` visited during sampling.)EOF")
+      .def_property_readonly(
+          "log_values",
+          [](const MCResult &self) {
+            assert(self.log_values.rows() % self.n_chains == 0);
+            return detail::as_readonly(py::array_t<Complex, py::array::c_style>{
+                {self.log_values.rows() / self.n_chains, self.n_chains},
+                self.log_values.data(),
+                py::none()});
+          },
+          py::return_value_policy::reference_internal,
+          R"EOF(An array of `complex128` representing `Ψ(vᵢ)` for all
+                sampled visible configurations `vᵢ`.)EOF");
+
+  subm.def(
+      "compute_samples",
+      [](AbstractSampler &sampler, Index n_samples, Index n_discard) {
+        // Helper types and lambda for writing the shapes in a clean way:
+        using Shape2 = std::array<std::size_t, 2>;
+        using Shape3 = std::array<std::size_t, 3>;
+        const auto _ = [](Index i) { return static_cast<std::size_t>(i); };
+
+        MCResult result =
```

Comment:
MCResult is still used here though?

Reply:
As I said, it is not used in the Python interface. Removing it in C++ requires changing a lot of code without immediate benefit.